oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java

```diff
@@ -0,0 +1,513 @@
+        .add("audience", audience)
+        .add("transportFactoryClassName", transportFactoryClassName)
+        .add("quotaProjectId", quotaProjectId)
+        .toString();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof ExternalAccountAuthorizedUserCredentials)) {
+      return false;
+    }
+    ExternalAccountAuthorizedUserCredentials credentials =
+        (ExternalAccountAuthorizedUserCredentials) obj;
+    return super.equals(credentials)
+        && Objects.equals(this.clientId, credentials.clientId)
+        && Objects.equals(this.clientSecret, credentials.clientSecret)
+        && Objects.equals(this.refreshToken, credentials.refreshToken)
+        && Objects.equals(this.tokenUrl, credentials.tokenUrl)
+        && Objects.equals(this.tokenInfoUrl, credentials.tokenInfoUrl)
+        && Objects.equals(this.revokeUrl, credentials.revokeUrl)
+        && Objects.equals(this.audience, credentials.audience)
+        && Objects.equals(this.transportFactoryClassName, credentials.transportFactoryClassName)
+        && Objects.equals(this.quotaProjectId, credentials.quotaProjectId);
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Returns external account authorized user credentials defined by JSON contents using the format
+   * supported by the Cloud SDK.
+   *
+   * @param json a map from the JSON representing the credentials
+   * @param transportFactory HTTP transport factory, creates the transport used to get access tokens
+   * @return the external account authorized user credentials defined by the JSON
+   */
+  static ExternalAccountAuthorizedUserCredentials fromJson(
+      Map<String, Object> json, HttpTransportFactory transportFactory) throws IOException {
+    String audience = (String) json.get("audience");
+    String refreshToken = (String) json.get("refresh_token");
+    String tokenUrl = (String) json.get("token_url");
+    String tokenInfoUrl = (String) json.get("token_info_url");
+    String revokeUrl = (String) json.get("revoke_url");
+    String clientId = (String) json.get("client_id");
+    String clientSecret = (String) json.get("client_secret");
+    String quotaProjectId = (String) json.get("quota_project_id");
+
+    return ExternalAccountAuthorizedUserCredentials.newBuilder()
+        .setAudience(audience)
+        .setRefreshToken(refreshToken)
+        .setTokenUrl(tokenUrl)
+        .setTokenInfoUrl(tokenInfoUrl)
+        .setRevokeUrl(revokeUrl)
+        .setClientId(clientId)
+        .setClientSecret(clientSecret)
+        .setRefreshToken(refreshToken)
+        .setHttpTransportFactory(transportFactory)
+        .setQuotaProjectId(quotaProjectId)
+        .build();
+  }
+
+  private void readObject(ObjectInputStream input) throws IOException, ClassNotFoundException {
+    input.defaultReadObject();
+    transportFactory = newInstance(transportFactoryClassName);
+  }
+
+  private boolean canRefresh() {
```

Comment:
MArk as VisibleForTesting and cover?

Reply:
This is sufficiently tested through the public API surface.